From cdc9dd2e9c2454259394d4b7f46c9bb720db6643 Mon Sep 17 00:00:00 2001
Date: Fri, 22 Jan 2021 16:25:03 +0800
Subject: set vm.vendor by configure

Summary: <vendor>: <backout the vendor patch>
LLT: java -XshowSettings:properties
Bug url: NA
---
 hotspot/src/share/vm/runtime/vm_version.cpp | 7 ++++++-
 jdk/src/share/native/java/lang/System.c     | 6 +++---
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/hotspot/src/share/vm/runtime/vm_version.cpp b/hotspot/src/share/vm/runtime/vm_version.cpp
index c6a559521..5ee3a7942 100644
--- a/hotspot/src/share/vm/runtime/vm_version.cpp
+++ b/hotspot/src/share/vm/runtime/vm_version.cpp
@@ -142,7 +142,12 @@ const char* Abstract_VM_Version::vm_name() {
 
 
 const char* Abstract_VM_Version::vm_vendor() {
-  return "Huawei Technologies Co., Ltd";
+#ifdef VENDOR
+  return VENDOR;
+#else
+  return JDK_Version::is_gte_jdk17x_version() ?
+    "Oracle Corporation" : "Sun Microsystems Inc.";
+#endif
 }
 
 
diff --git a/jdk/src/share/native/java/lang/System.c b/jdk/src/share/native/java/lang/System.c
index 758cfabb3..ff80b0abd 100644
--- a/jdk/src/share/native/java/lang/System.c
+++ b/jdk/src/share/native/java/lang/System.c
@@ -110,13 +110,13 @@ Java_java_lang_System_identityHashCode(JNIEnv *env, jobject this, jobject x)
 
 /* Third party may overwrite these values. */
 #ifndef VENDOR
-#define VENDOR "Huawei Technologies Co., Ltd"
+#define VENDOR "Oracle Corporation"
 #endif
 #ifndef VENDOR_URL
-#define VENDOR_URL "http://jdk.rnd.huawei.com/"
+#define VENDOR_URL "http://java.oracle.com/"
 #endif
 #ifndef VENDOR_URL_BUG
-#define VENDOR_URL_BUG "http://jdk.rnd.huawei.com/"
+#define VENDOR_URL_BUG "http://bugreport.sun.com/bugreport/"
 #endif
 
 #define JAVA_MAX_SUPPORTED_VERSION 52
-- 
2.19.0

